From 8b34d272684c6d624ff5e2ba23795067a3dbe2f4 Mon Sep 17 00:00:00 2001
From: Alan Winkowski <walan@marvell.com>
Date: Tue, 20 Nov 2018 16:36:37 +0200
Subject: [PATCH 0798/1051] net: mvpp2: fix: set default engine in C3Hx lookups

It is recommended to set a default engine for the default C3Hx lookups.
The engine can be overridden afterwards when setting the HEK fields

Change-Id: Ifb404b089f3a60e7de97545c2507460104f415c9
Signed-off-by: Alan Winkowski <walan@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/61113
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Yan Markman <ymarkman@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c
index ab49757c36ff..84f7fea5fd61 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c
@@ -506,6 +506,7 @@ static void mvpp2_cls_flow_init(struct mvpp2 *priv, struct mvpp2_cls_flow *flow)
 		memset(&fe, 0, sizeof(fe));
 		fe.index = MVPP2_PORT_FLOW_HASH_ENTRY(i, flow->flow_id);
 
+		mvpp2_cls_flow_eng_set(&fe, MVPP22_CLS_ENGINE_C3HA);
 		mvpp2_cls_flow_port_id_sel(&fe, true);
 		mvpp2_cls_flow_pri_set(&fe, i + 1);
 		mvpp2_cls_flow_port_add(&fe, BIT(i));
-- 
2.17.1

